addons/common/functions/fnc_setBuildingMarker.sqf

```diff
@@ -0,0 +1,64 @@
+#include "script_component.hpp"
+/*
+ * Author: Ampersand
+ * Addes or removes a rectangle marker on object bounding box, to make placed building look like terrain building. Called by CBA serverEvent.
+ *
+ * Arguments:
+ * 0: Object <OBJECT>
+ * 1: Set Marker <BOOLEAN>
+ *
+ * Return Value:
+ * Marker name <STRING>
+ *
+ * Example:
+ * [_object, true] call zen_common_fnc_setbuildingMarker
+ *
+ * Public: No
+ */
+
+params ["_object", "_setMarker"];
+
+if !(_object isEqualType objNull) exitWith {""};
+
+private _marker = _object getVariable [QGVAR(buildingMarker), ""];
+private _hasMarker = _marker isNotEqualTo "";
+
+if (_setMarker) then {
+    // Update marker
+    if (_hasMarker) exitWith {
+        _marker setMarkerPosLocal getPos _object;
+        _marker setMarkerDir getDir _object;
+
+        _marker
+    };
+
+    // Create marker
+    _marker = createMarker [format ["%1_%2", QGVAR(buildingMarker), _object call bis_fnc_netId], _object];
+    _marker setMarkerShapeLocal "RECTANGLE";
+    _marker setMarkerSizeLocal ((0 boundingBoxReal _object) select 1);
+    _marker setMarkerDirLocal getDir _object;
+    _marker setMarkerColorLocal "ColorGrey";
+    _marker setMarkerBrush "SolidFull";
+
+    // Tracking
+    _object setVariable [QGVAR(buildingMarker), _marker, true];
+    private _id = _object addEventHandler ["Deleted", {
+        params ["_entity"];
+        private _marker = _entity getVariable [QGVAR(buildingMarker), ""];
+        deleteMarker _marker;
+    }];
+    _object setVariable [QGVAR(buildingMarkerDeletedEHID), _id, true];
```

Comment:
If this is only needed on the server, then it doe snot need to be broadcasted.

Reply:
https://github.com/zen-mod/ZEN/pull/704/commits/df7388c87d42a6f409e1256b36478aa186d5e9f1#diff-904878a4c3022a6347e5b85fe824f3641b2911718a3985159b1d6bdf0b3ad5e0R49-R66